### ISSUE TITLE
Parallel recovery for nodePool with 2 replicas

### ISSUE THREAD

**Issue by saketmht** [Opster/opensearch-k8s-operator#444]
Config - 
```
  nodePools:
    - component: masters
      replicas: 2
      resources:
         requests:
            memory: "1Gi"
      roles:
        - "data"
        - "cluster_manager"
      diskSize: "1Gi"
```
When I delete only one pod, the pod starts up again and the 2 pods are able to form quorum again. But if I delete both the pods, only one pod comes up at a time as sts has `OrderedReady` podManagementPolicy so the single pod is not able to form a quorum. The parallel recovery should also kick in in this case and allow both the pods to come up together.

**Comment by saketmht** [comment]
But the operator doesn't explicitly disallow down-scaling clusters right? There can be a use-case where a user had to temporarily scale down the cluster to 2 replicas and something went wrong during that time.

**Comment by swoehrl-mw** [comment]
@saketmht There is no technical limitation in the operator on having less than 3 nodes, correct. But with less than 3 it will not behave correctly.

>  There can be a use-case where a user had to temporarily scale down the cluster to 2 replicas and something went wrong during that time.

I'd consider that an edge case not worthy of extra logic. 
Without having checked the code or tested it: If you had 3 replicas deployed (so the PVCs are there), scaled down to 2, lost one, and then changed replicas back to 3 it should start the recovery.

**Comment by saketmht** [comment]
@swoehrl-mw I don't think it requires any extra logic. The current parallel recovery code should be more than enough.

https://github.com/Opster/opensearch-k8s-operator/blob/4edd47a61f05e2a3b2e38d826f9fb5ca33e55f1f/opensearch-operator/pkg/reconcilers/cluster.go#L179

Changing this to 
```
if pvcCount >= int(nodePool.Replicas) && existing.Status.ReadyReplicas < nodePool.Replicas-1 { 
```
should do the job and I think comparing with `ReadyReplicas` is actually more accurate than just the `Replicas` count.

**Comment by IshaGirdhar** [comment]
@idanl21  @saketmht I tested with replacing Replicas to ReadyReplicas, and it works fine and with 2 replicas, pods are able to come back up, I tried few other scenarios as well to make sure it doesn't break any existing path. 
1. Scaling up and scaling down after recovery for 2 replicas
2. Trying recovery for more than 2 replicas
Is there anything specific which you would think should be tested before I raise a PR?
